test/simulator_tests/birth_death_simulator_test.py

```diff
@@ -0,0 +1,573 @@
+import unittest
+
+import networkx as nx
+import numpy as np
+
+from cassiopeia.simulator.BirthDeathFitnessSimulator import (
+    BirthDeathFitnessSimulator,
+)
+from cassiopeia.simulator.BirthDeathFitnessSimulator import (
+    BirthDeathFitnessError,
+)
+
+
+def get_leaves(tree):
+    return [n for n in tree.nodes if tree.out_degree(n) == 0]
+
+
+def test_tree(tree: nx.DiGraph):
+    """A helper function for testing simulated trees.
+
+    Outputs the (independently calculated) total lived time for each extant
+    lineage, the number of extant lineages, and whether the tree has the
+    expected node degrees (to ensure unifurcation collapsing was done
+    correctly).
+    """
+    tree = tree.copy()
+    tree.nodes[0]["total_time"] = 0
+    for i in nx.edge_dfs(tree):
+        tree.nodes[i[1]]["total_time"] = (
+            tree.nodes[i[0]]["total_time"] + tree.edges[i]["weight"]
+        )
+
+    leaves = get_leaves(tree)
+    times = []
+    for i in leaves:
+        times.append(tree.nodes[i]["total_time"])
+
+    out_degrees = [tree.out_degree(n) for n in tree.nodes]
+    out_degrees.pop(0)
+
+    correct_degrees = all(x == 2 or x == 0 for x in out_degrees)
+
+    return times, len(leaves), correct_degrees
+
+
+class BirthDeathSimulatorTest(unittest.TestCase):
+    def test_remove_and_prune_lineage_all(self):
+        bd_sim = BirthDeathFitnessSimulator()
+        tree = nx.balanced_tree(2, 3, create_using=nx.DiGraph())
+        leaves = get_leaves(tree)
+        for i in leaves:
+            bd_sim.remove_and_prune_lineage(i, tree)
```

Comment:
This is a good example for why we'd like to have this function in the `data.utilities` module -- it's super useful and doesn't make sense to have to create a `bd_sim` object to use it